bee-common/bee-common-derive/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use proc_macro::{self, TokenStream};
+use proc_macro_error::{abort, proc_macro_error};
+use quote::{quote, ToTokens};
+use syn::{parse_macro_input, Data, DeriveInput, Type};
+
+mod packable;
+
+#[proc_macro_error]
+#[proc_macro_derive(Packable, attributes(packable))]
+pub fn packable(input: TokenStream) -> TokenStream {
+    let DeriveInput {
+        ident,
+        data,
+        attrs,
+        generics,
+        ..
+    } = parse_macro_input!(input);
+    // Check if the type has an `error` attribute.
+    let error_type = packable::parse_attr::<Type>("error", &attrs).map(ToTokens::into_token_stream);
+
+    match data {
+        Data::Struct(data) => {
+            // Use `Infallible` if there was no error attribute.
+            let error_type = error_type.unwrap_or_else(|| quote!(core::convert::Infallible));
+            // Generate the implementation for the struct.
+            let (pack, packed_len, unpack) = packable::gen_bodies_for_struct(data.fields);
+            packable::gen_impl(&ident, &generics, error_type, pack, unpack, packed_len).into()
```

Comment:
OCD triggered: `pack, packed_len, unpack` not in the same order as passed to `gen_impl`. Preferred order: `packed_len, pack, unpack`.